### PR TITLE
webpack dynamic require fixed

### PR DESCRIPTION
**Pull request by psrebniak**
Fixed dynamic require call elastic/apm-nodejs-http-client#43

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -69,7 +69,20 @@ function Client (opts) {
   this.sent = 0 // number of events written to the socket
   this._active = false
   this._onflushed = null
-  this._transport = require(opts.serverUrl.protocol.slice(0, -1)) // 'http:' => 'http'
+  this._transport = null
+  switch(opts.serverUrl.protocol.slice(0, -1)) { // 'http:' => 'http'
+    case 'http': {
+      this._transport = require('http')
+      break;
```
Comment (watson) [inline index.js]: Our linter doesn't like semicolons:

```suggestion
      break
```

File: index.js [hunk 2]
```diff
@@ -69,7 +69,20 @@ function Client (opts) {
...
+    }
+    case 'https': {
+      this._transport = require('https')
+      break;
```
Comment (watson) [inline index.js]: Our linter doesn't like semicolons:

```suggestion
      break
```

File: index.js [hunk 3]
```diff
@@ -69,7 +69,20 @@ function Client (opts) {
...
+    }
+    default: {
+      throw new Error('Unknown protocol ' + opts.serverUrl.protocol.slice(0, -1))
+    }
+  }
```
Comment (watson) [inline index.js]: Our linter requires an empty line above and below the switch statement:

```suggestion
  }

```

File: index.js [hunk 4]
```diff
@@ -69,7 +69,20 @@ function Client (opts) {
   this.sent = 0 // number of events written to the socket
   this._active = false
   this._onflushed = null
-  this._transport = require(opts.serverUrl.protocol.slice(0, -1)) // 'http:' => 'http'
+  this._transport = null
```
Comment (watson) [inline index.js]: Our linter requires an empty line above and below the switch statement:

```suggestion
  this._transport = null

```

**Comment by psrebniak** [comment]
@watson all fixed

**Comment by psrebniak** [comment]
@watson should I squash and rebase it or it is not necessary?

**Comment by watson** [comment]
Nope, I can just do that when I merge 👍

**Comment by watson** [comment]
This will be included in the next release of the Node.js APM agent

**Pull request merged**